src/inmanta/references.py

```diff
@@ -169,6 +169,33 @@ def get_arg_value(
         return resource.get_reference_value(self.id, logger)
 
 
+class MutatedJsonArgument(Argument):
+    """
+    Json-like argument to a reference that contains reference in the json structure
```

Comment:
```suggestion
    Json-like argument that contains another reference in the json structure
```
Right? The reference has nothing to do with it? Or does it?